pkgs/tools/package-management/appimage-run/test.nix

```diff
@@ -11,21 +11,27 @@ let
   };
 in
   runCommand "appimage-run-tests" {
-    buildInputs = [ appimage-run glibcLocales file ];
+    buildInputs = [ appimage-run glibcLocales file xdg-utils ];
     meta.platforms = [ "x86_64-linux" ];
   }
   ''
     export HOME=$(mktemp -d)
     set -x
+
     # regression test for #101137, must come first
     LANG=fr_FR appimage-run ${sample-appImage} --list ${sample-appImage}
+
     # regression test for #108426
     cp ${sample-appImage} foo.appImage
     LANG=fr_FR appimage-run ${sample-appImage} --list foo.appImage
     cp ${owdtest} owdtest.AppImage.gz
     gunzip owdtest.AppImage.gz
     appimage-run owdtest.AppImage
+
+    # Verify desktop entry
+    XDG_DATA_DIRS="${appimage-run}/share"
+    [[ "$(xdg-mime query default application/vnd.appimage)" == '${appimage-run.name}.desktop' ]]
```

Comment:
Would this not be a preference of the user? Or is this strictly required by the application?

Reply:
I think the preference only comes in once there are more than one application for a MIME type. Otherwise the user would have the tedious job of assigning defaults to every single file type, even when there's an application already in place to handle it.